libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1134,6 +1150,17 @@ void AP_OSD_Screen::draw_altitude(uint8_t x, uint8_t y)
     backend->write(x, y, false, "%4d%c", (int)u_scale(ALTITUDE, alt), u_icon(ALTITUDE));
 }
 
+void AP_OSD_Screen::draw_avgvolt(uint8_t x, uint8_t y)
+{
+    AP_BattMonitor &battery = AP::battery();
+    uint8_t pct = battery.capacity_remaining_pct();
+    uint8_t p = (100 - pct) / 16.6;
+    float v = battery.voltage();
+    uint8_t num_cells = (uint8_t)((v/4.3f) + 1);
```

Comment:
Sorry, but there is no way to detect the number of cells of a battery based on its voltage.....detection would have to know the type (NiCad/NimH/LiFe/Lipo/LiIon,etc.), the state of charge, the internal resistance, and the present current draw.....you could add an OSD parameter for number of cells, but then this panel would not be adding any user information and just consuming code space needlessly...we do have a panel for displaying the directly measured cell voltage from a sensor in MSP systems

Reply:
This is true, but this is a useful approximation for the vast majority of users, who use either Li-Ion or LiPo, so I believe it's a worthwhile addition. Rejecting this PR is certainly fine, but it feels a bit like saying "if it's not useful to everyone, it shouldn't be useful to anyone".

Reply:
I disagree, even with LiPo only the useful voltage ranges overlap for each cell number...12.2V on 4s is within 3S, etc.   and LiIon especially have large voltage sags under load and operate well below 3.0v ....
the issue is not if its useful for everyone, but does it easily create problems for many users....and I believe it will incorrectly display and warn in many siturations...

Reply:
I edited my comment after you had commented, I agree that detection should be stable, but I don't know how to detect the number of cells at powerup and store that somewhere.